source/sigv4.c

```diff
@@ -966,9 +985,9 @@ static SigV4Status_t lowercaseHexEncode( const SigV4String_t * pInputStr,
     {
         for( i = 0; i < pInputStr->dataLen; i++ )
         {
-            *hex = digitArr[ ( pInputStr->pData[ i ] & 0xF0 ) >> 4 ];
+            *hex = digitArr[ ( bytes[ i ] & 0xF0U ) >> 4 ];
             hex++;
-            *hex = digitArr[ ( pInputStr->pData[ i ] & 0x0F ) ];
+            *hex = digitArr[ ( bytes[ i ] & 0x0FU ) ];
```

Comment:
Why do we need an additional variable here? Just preference?

Reply:
It's since MISRA requires all the operands of bitwise operators to be unsigned, so I had to either add an unsigned variable or do `( ( ( uint8_t ) pInputStr->pData[ i ] ) & 0x0FU )`.